### PR TITLE
Add non-parametric survival

### PR DESCRIPTION
**Pull request by FBartos**
This has all the main functions for non-parametric survival analysis.
(also some unused code for semi-parametric etc should be reviewed separately later)

There are some things that need resolving:

- [ sidestepped ] the survival figures (lines 264-281 with TODO's explaining the issues)
- [x] missing icon for survival (someone who likes playing with images should make that)

In QML
- [x] make the event indicator to be the second lvl of the event status variable by default
- [ not possible ] make "lifeTableStepsTo" correspond to the maximum time
- [ not possible ] make "lifeTableStepsSize" correspond to the maximum time/10

(there is a simple example `leukemia.csv` data set in the `tests/` folder)

### CODE REVIEW

File: inst/qml/NonParametricSurvivalAnalysis.qml [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+			suggestedColumns:	["nominal"]
+			singleVariable:		true
+		}
+
+		DropDown
+		{
+			name:				"eventIndicator"
+			label:				qsTr("Event Indicator")
+			source:				[{name: "eventStatus", use: "levels"}]
+		}
```
Comment (vandenman) [inline inst/qml/NonParametricSurvivalAnalysis.qml]:
```suggestion
			source:				[{name: "eventStatus", use: "levels"}]
			indexDefaultValue:  eventStatusId ? 1 : 0
		}
```

File: inst/qml/NonParametricSurvivalAnalysis.qml [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+		{
+			name:				"timeToEvent"
+			title:				qsTr("Time to Event")
+			suggestedColumns:	["scale"]
+			singleVariable:		true
+		}
+
+		AssignedVariablesList
+		{
+			name:				"eventStatus"
```
Comment (vandenman) [inline inst/qml/NonParametricSurvivalAnalysis.qml:44]:
```suggestion
			id:					eventStatusId
			name:				"eventStatus"
```

**Comment by JorisGoosen** [comment]
Ok, If this PR gets merged I can add it to `development`

Let me know ;)

File: inst/Description.qml [hunk 3]
```diff
@@ -3,12 +3,45 @@ import JASP.Module	1.0
...
+	version:		"0.18.1"
+	author:			"JASP Team"
+	maintainer:		"JASP Team <info@jasp-stats.org>"
+	website:		"jasp-stats.org"
+	license:		"GPL (>= 2)"
+
+	GroupTitle
+	{
+		title:	qsTr("Classical")
+		icon:	"survival-analysis.svg"
```
Comment (Kucharssim) [inline inst/Description.qml:20]: nice icon!

File: inst/qml/NonParametricSurvivalAnalysis.qml [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+		}
+	}
+
+	Group
+	{
+
+		CheckBox
+		{
+			name:	"tests"
+			label:	qsTr("Tests")
```
Comment (Kucharssim) [inline inst/qml/NonParametricSurvivalAnalysis.qml]: Is it necessary to nest the individual tests under the `tests` checkbox? Why not just use a `Group{}`? This feels a bit convoluted to me.

File: inst/qml/NonParametricSurvivalAnalysis.qml [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+				min:			lifeTableStepsFrom.value
+				visible:		lifeTableStepsType.value == "fixedSize"
+			}
+		}
+	}
+
+	CheckBox
+	{
+		name:	"survivalCurvePlot"
+		label:	qsTr("Survival curve plot")
```
Comment (Kucharssim) [inline inst/qml/NonParametricSurvivalAnalysis.qml:169]: The risk table, cumulative events, censoring plots, etc. are nested under survival curve plot despite generating independent plots. Is that on purpose? Could we put it under a `Group`/`Section` titled "Plots" and have them as separate options?

**Review by Kucharssim — CHANGES_REQUESTED** [review]
Looks quite good, the functionality is not that complex in the end, so reviewing this was not as difficult as I feared 😅 I also played around with the analysis a bit in JASP and it seems to run fine.

I just have some questions about organizing the input form, see code comments.

Otherwise:

1. Since this repo was created the tests workflow changed. Could you please update the tests workflow from https://github.com/jasp-stats/jaspModuleTemplate/blob/master/.github/workflows/unittests.yml in here so that tests can run on GitHub? 
2. Could you add the i18nCheck and translations workflow? https://github.com/jasp-stats/jaspModuleTemplate/blob/master/.github/workflows/i18nCheck.yml and https://github.com/jasp-stats/jaspModuleTemplate/blob/master/.github/workflows/translations.yml
3. Can you write up some tests please?

File: inst/qml/NonParametricSurvivalAnalysis.qml (hunk 5 above)
Comment (FBartos) [reply inst/qml/NonParametricSurvivalAnalysis.qml:169]: Yeah, this one is actually on purpose -- the final figure should be a one panel, but we were having trouble aligning the x-axis etc, properly. Hopefully, we could add a checkbox to the option that merges them together (as supposed to) in the next version.

File: inst/qml/NonParametricSurvivalAnalysis.qml (hunk 4 above)
Comment (FBartos) [reply inst/qml/NonParametricSurvivalAnalysis.qml]: fixed

**Comment by FBartos** [comment]
I will add the tests (and worked example tomorrow -- I have a textbook at home, so we can also verify against that)

**Comment by FBartos** [comment]
ok, unit tests are in, let's merge it if you agree

**Review by Kucharssim — APPROVED** [review]

**Pull request merged**